builder/frameworks/espidf.py

```
@@ -195,177 +197,193 @@ def _get_installed_standard_pip_packages():
+                    if response.ok:
+                        return response.content.decode('utf-8')
+                except Exception as e:
+                    print(f"Error downloading {file_entry}: {e}")
                     return ""
-                return target
-            if "file://" in file:
-                file_path = join(PROJECT_DIR,file.lstrip("file://").split(os.path.sep)[-1])
+            
+            # Handle local files
+            if "file://" in file_entry:
+                file_path = join(PROJECT_DIR, file_entry.lstrip("file://").split(os.path.sep)[-1])
```

Ruff (0.11.9)

Using `.strip()` with multi-character strings is misleading

(B005)